routes/users.js

```diff
@@ -1,9 +1,27 @@
 var express = require('express');
 var router = express.Router();
+const { createUser } = require('../services/user-service');
+const ApiError = require('../utils/api-error');
 
-/* GET users listing. */
-router.get('/', function(req, res, next) {
-  res.send('respond with a resource');
+
+/* Create a new user */
+router.post('/register', async function (req, res, next) {
+  const { name, email, password } = req.body;
+  try {
+    const token = await createUser({ name, email, password });
+    res.status(201).json({ success: true, message: "User registered successfully", token });
+
+  } catch (err) {  
+    if (err instanceof ApiError) {
+      res.status(err.statusCode).json({
+        status: 'error',
+        message: err.message,
+        errors: err.errors,
+      });
+    } else {
+      res.status(500).json({ status: 'error', message: 'Server error' });
```

Comment:
const apierror = new ApiError(msg || 'Internal Server Error', 500);
maybe log the error
next(apierror)